### DISCUSSION TITLE
Timeout error when saving large file to Google Drive

### DISCUSSION THREAD

**Discussion by brnpl89** [jgraph/drawio#3821] [General]
We are running version 21.0.2 self-hosted and we are experiencing this issue with diagrams that are large in byte size (more than 1.5MB) or too complex. When we open a large diagram file from Google Drive, make some changes and want to save them (whether through automatic save after making a change or manual change through "File -> Save"), we get a timeout error (as evidenced by the screenshots below). 

![image](https://github.com/jgraph/drawio/assets/143200688/2ffcd3e2-46b8-4f4c-83c3-b9c1101871b7)

![image](https://github.com/jgraph/drawio/assets/143200688/a97fad97-48e8-4357-bcd5-8a962f5c93ba)

![image](https://github.com/jgraph/drawio/assets/143200688/02eb3e7a-79ba-48b3-850d-e7b4ab84bed6)

We would like to know if this timeout comes from DrawIO or Google Drive itself. If it comes from DrawIO, we want to know if it’s possible to change the value of the timeout and where exactly we can do that.

We opened an issue for this before (https://github.com/jgraph/drawio/issues/3815) but since nobody replied, we were hoping to get some answers here.

**Comment by brnpl89** [comment]
Also, we forgot to mention that if we disable the "Compressed" property in the menu, this issue doesn't happen.

**Comment by brnpl89** [comment]
On the DriveClient.js file, i located the DriveClient.prototype.createUploadRequest function.

If i put a setTimeout on the request object (reqObj) inside that function, will that fix this issue?

**Comment by davidjgraph** [comment]
If compression is the issue, just use the latest build. Compression was removed at some point.

**Comment by Dubrzr** [comment]
I have the same problem on very slow internet connexion, I activated compression and cannot compress my file more, what should I do? How can I increase the timeout delay?

Thank you!

**Reply by davidjgraph** [reply]
If your connection is too slow to use a SaaS tool, use the desktop version, but remember to back files up.

**Reply by Dubrzr** [reply]
Well I use many other SaaS tools, edit Powerpoints online and stuff like that and it works well even with my 0.8mbps upload speed, but not draw.io, wondering if the timeout is configured in draw.io or on Google drive side?

**Reply by davidjgraph** [reply]
The save is directly to Google Drive, it doesn't go via our servers. How big is the file?

**Reply by Dubrzr** [reply]
I'm not on that slow internet connection anymore, but the file i wanted to edit back then was 10.6MB.

**Comment by connor42312** [comment]
I am experiencing the same issue with the message "Unsaved changes, click here to save. (Timeout)". File size is 16.1MB.  Been monitoring my internet connection and it looks fine (no disconnects, up to 235 MB/s upload and 4 MB/s download, 24ms latency). Tested both on the desktop chrome app as well as in chrome web browser. Have tried reauthenticating with my google account and also rebooting my machine. Is the upload speed too slow at 4 MB/s or is the file size too big? Something else I can look into?

![image](https://github.com/user-attachments/assets/8a8c6c7a-ffba-4dcf-904a-60b0a7650c1c)

**Comment by DaanRosendal** [comment]
I encountered the same issue. This appears to be caused by low upload speeds, as when I switched to a faster connection at my university, the problem disappeared.
